src/sapling/saplingscriptpubkeyman.cpp

```diff
@@ -583,6 +583,12 @@ CAmount SaplingScriptPubKeyMan::GetOutPointValue(const CWalletTx& tx, const Sapl
     return tx.mapSaplingNoteData.at(op).amount ? *(tx.mapSaplingNoteData.at(op).amount) : 0;
 }
 
+Optional<std::string> SaplingScriptPubKeyMan::GetOutPointMemo(const CWalletTx& tx, const SaplingOutPoint& op)
+{
+    auto it = tx.mapSaplingNoteData.find(op);
+    return it != tx.mapSaplingNoteData.end() && it->second.memo ? Optional<std::string>(std::string(it->second.memo->begin(), it->second.memo->end())) : nullopt;
```

Comment:
should probably be sanitizing this string, otherwise you get a ton of invalid characters tailing the actual memo text.

```diff
Index: src/sapling/saplingscriptpubkeyman.cpp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/sapling/saplingscriptpubkeyman.cpp	(revision 759d757b0a2072a680c6898a4984a40d011e2631)
+++ src/sapling/saplingscriptpubkeyman.cpp	(date 1606813243868)
@@ -5,6 +5,7 @@
 
 #include "sapling/saplingscriptpubkeyman.h"
 #include "chain.h" // for CBlockIndex
+#include "utilstrencodings.h" // for SanitizeString()
 #include "validation.h" // for ReadBlockFromDisk()
 
 void SaplingScriptPubKeyMan::AddToSaplingSpends(const uint256& nullifier, const uint256& wtxid)
@@ -586,7 +587,7 @@
 Optional<std::string> SaplingScriptPubKeyMan::GetOutPointMemo(const CWalletTx& tx, const SaplingOutPoint& op)
 {
     auto it = tx.mapSaplingNoteData.find(op);
-    return it != tx.mapSaplingNoteData.end() && it->second.memo ? Optional<std::string>(std::string(it->second.memo->begin(), it->second.memo->end())) : nullopt;
+    return it != tx.mapSaplingNoteData.end() && it->second.memo ? Optional<std::string>(SanitizeString(std::string(it->second.memo->begin(), it->second.memo->end()))) : nullopt;
 }
 
 Optional<std::pair<

```

---------
without sanitizing the string, this is what it looks like for me:
![ScreenShot_20201201013021](https://user-images.githubusercontent.com/7393257/100721952-e0763c00-3374-11eb-88fc-48f9562af1ec.png)